04.ls/ls.rb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+COLUMN_NUM = 3
+
+def main
+  entries = list_entries
+  print_entries(entries)
+end
+
+def list_entries
+  Dir.glob('*').map do |file|
+    File.basename(file)
+  end
+end
+
+def print_entries(entries)
+  entries_table = create_table(entries)
+  max_tab_num = get_tab_num(entries.max_by(&:length)) + 1
+
+  entries_table.each do |row_entries|
+    row_entries.each do |entry|
+      delimiter = "\t" * (max_tab_num - get_tab_num(entry))
+      printf("%s#{delimiter}", entry)
```

Comment:
```suggestion
      print "#{entry}#{delimiter}"
```

好みの問題ですが、こっちの方がシンプルかつ直感的で理解しやすい気がします